tests/compatibility-test.py

```diff
@@ -263,106 +267,36 @@ def get_new_value():
         client.close()
 
     def test_detached_actor(self):
-        # This test will run a ray client and start a detached actor at first.
-        # Then we will kill the head node and kuberay will start a new head node
-        # replacement. Finally, we will try to connect to the detached actor again.
         client = docker.from_env()
         container = client.containers.run(ray_image, remove=True, detach=True, stdin_open=True, tty=True,
-                                          network_mode='host', command=["/bin/sh", "-c", "python"])
-        s = container.attach_socket(
-            params={'stdin': 1, 'stream': 1, 'stdout': 1, 'stderr': 1})
-        s._sock.setblocking(0)
-        s._sock.sendall(b'''
-import ray
-import time
+                                            network_mode='host', command=["/bin/sh"])
+        ray_namespace = ''.join(random.choices(string.ascii_lowercase, k=10))
+        logger.info(f'namespace: {ray_namespace}')
 
-def retry_with_timeout(func, count=90):
-    tmp = 0
-    err = None
-    while tmp < count:
-        try:
-            return func()
-        except Exception as e:
-            err = e
-            tmp += 1
-    assert err is not None
-    raise err
+        # Register a detached actor
+        utils.copy_to_container(container, 'tests/scripts', '/usr/local/', 'test_detached_actor_1.py')
+        exit_code, output = utils.exec_run_container(container, f'python3 /usr/local/test_detached_actor_1.py {ray_namespace}', 180)
 
-ray.init(address='ray://127.0.0.1:10001')
-
-@ray.remote
-class A:
-    def ready(self):
-        import os
-        return os.getpid()
-
-a = A.options(name="a", lifetime="detached", max_restarts=-1).remote()
-res1 = ray.get(a.ready.remote())
-print('ready')
-
-        ''')
-
-        count = 0
-        while count < 90:
-            try:
-                buf = s._sock.recv(4096)
-                logger.info(buf.decode())
-                if buf.decode().find('ready') != -1:
-                    break
-            except Exception as e:
-                pass
-            time.sleep(1)
-            count += 1
-        if count >= 90:
-            raise Exception('failed to run script')
+        if exit_code != 0:
+            raise Exception(f"Exception happens during the execution of test_detached_actor_1.py. The exit code is {exit_code}.")
```

Comment:
It will be printed in the `exec_run_container` function.

```python
def exec_run_container(container, cmd, timeout_sec, silent = False):
    timeout_cmd = 'timeout {}s {}'.format(timeout_sec, cmd)
    # If the exit_code is 124, 125, 126, 127, 137, it is related to the `timeout` command.
    # See https://manpages.courier-mta.org/htmlman1/timeout.1.html for more details.
    exit_code, output = container.exec_run(cmd = timeout_cmd)
    if not silent:
        logger.info(f"cmd: {timeout_cmd}")
        logger.info(f"exit_code: {exit_code}")
        logger.info(f"output: {output.decode()}")
    return exit_code, output.decode()
```

Reply:
Ah, great point! Maybe the exception could say something like "See above for command output"?
You could also add a doc string to this method saying that the method executes the command and logs the output.

Reply:
Updated the exception message and added the doc string.